# Introduction

We are using a new protocol for sequencing the HAMBI microbial community that allows for a greater degree of multiplexing by preparing libraries in-house at the University of Turku [Center for Evolutionary Applications](https://www.utu.fi/en/university/faculty-of-science/biology/center-of-evolutionary-applications). The protocol follows the [Adapterama series II](https://peerj.com/articles/7786/). Fusion primers compatible with Illumina iTru primers can be designed using a simple spreadsheet from the supplementary material of the paper. These fusion primers have quadruple indexes - one pair of standard 8 bp Illumina iTru indexes and one pair of custom 5 bp indexes.

# Demultiplexing
For demultiplexing I have been using the pheniqs software ([https://doi.org/10.1186/s12859-021-04267-5](https://doi.org/10.1186/s12859-021-04267-5))that is [available from here](https://github.com/biosails/pheniqs). There documentation is available [here](https://biosails.github.io/pheniqs/).

The strategy I have used is to first demultiplex on the outer Illumina iTru indexes creating, for example, 6 different demultiplexed bam files as shown here. This is the [dim1](dim1) directory. Then these six bam files are each demultiplexed one-by-one using the pairs of the inner custom indexes in the [dim2](dim2) directory. The Adapteramma II indexes are 5 bp long, but 0-3 bp of spacer precede these indexes meaning that they don't all start at the same sequencing cycle in the read. The spacer sequence is included to create extra diversity at the read positions since these are low-diversity amplicon libraries. My solution has been to incorporate the spacer sequence and then 0-3 bp of the 5' sequence of the 16S primers to ensure that each index is 8 bp long and always starts on the first sequencing cycle.

The indexes for demultiplexing are in [samplesheet.tsv](samplesheet.tsv). The iTru indexes are self explanatory. The `inline_barcode_F_seq`, `inline_barcode_F_seq` show the 5 bp index in capital and the 0-3 bp of spacer in lower case. The `inline_barcode_F_8` and	`inline_barcode_R_8` are the 8 bp indexes starting on the same cycle that either incorporate the spacer sequence or the spacer and part of the 5' end of the `Bact-0341-b-S-17` and `S-D-Bact-0785-a-A-21` 16S primers from the paper.